src/tensorlake/function_executor/service.py

```diff
@@ -41,13 +44,87 @@
     InitializeResponse,
     InvocationStateRequest,
     InvocationStateResponse,
+    ListTasksRequest,
+    ListTasksResponse,
+    ProgressUpdate,
     RunTaskRequest,
     RunTaskResponse,
+    Task,
+    TaskFailureReason,
+    TaskOutcomeCode,
+    TaskResult,
 )
 from .proto.function_executor_pb2_grpc import FunctionExecutorServicer
 from .std_outputs_capture import flush_logs, read_till_the_end
 
 
+def _run_task_request_to_function_inputs(run_request: RunTaskRequest) -> FunctionInputs:
+    """Convert RunTaskRequest to FunctionInputs."""
+    request = FunctionInputs()
+
+    if run_request.HasField("function_input"):
+        request.function_input.CopyFrom(run_request.function_input)
+
+    if run_request.HasField("function_init_value"):
+        request.function_init_value.CopyFrom(run_request.function_init_value)
+
+    return request
+
+
+def _run_task_request_to_task(run_request: RunTaskRequest) -> Task:
+    """Convert RunTaskRequest to Task."""
+    return Task(
+        task_id=run_request.task_id,
+        namespace=run_request.namespace,
+        graph_name=run_request.graph_name,
+        graph_version=run_request.graph_version,
+        function_name=run_request.function_name,
+        graph_invocation_id=run_request.graph_invocation_id,
+        allocation_id=run_request.allocation_id,
+        request=_run_task_request_to_function_inputs(run_request),
+    )
+
+
+def _task_result_to_run_task_response(
+    task_id: str, result: TaskResult
+) -> RunTaskResponse:
+    """Convert TaskResult to RunTaskResponse."""
+    response = RunTaskResponse(
+        task_id=task_id,
+        function_outputs=list(result.function_outputs),
+        next_functions=list(result.next_functions),
+        stdout=result.stdout,
+        stderr=result.stderr,
+        is_reducer=result.is_reducer,
+        outcome_code=result.outcome_code,
+        failure_reason=result.failure_reason,
+    )
+
+    if result.HasField("metrics"):
+        response.metrics.CopyFrom(result.metrics)
+
+    if result.HasField("invocation_error_output"):
+        response.invocation_error_output.CopyFrom(result.invocation_error_output)
+
+    return response
+
+
+class _TaskExecution:
+    def __init__(self, task: Task):
+        self.task = task
+        self.thread: threading.Thread | None = None
+        # Initialize with empty result using TaskResult
+        self.task.result.CopyFrom(
+            TaskResult(
```

Comment:
Why not just None? Even from code lines perspective this is shorted and it's less code to maintain when something changes in TaskResult.

Reply:
Style.  :-). Done.